parse.go

```diff
@@ -1,29 +1,32 @@
 package main
 
 import (
+	"bytes"
 	"encoding/json"
-	"io/ioutil"
 	"os"
 
-	"bytes"
 	log "github.com/Sirupsen/logrus"
 )
 
 func ParseAnswers(path string) (out Versions, err error) {
-	var delta *MetadataDelta
 	var v Versions
-	data, err := ioutil.ReadFile(path)
+	var md MetadataDelta
+	f, err := os.Open(path)
 	if err != nil {
 		if os.IsNotExist(err) {
 			log.Warn("Failed to find: ", path)
 			return v, nil
 		}
 		return nil, err
 	}
+	defer f.Close()
```

Comment:
You always defer after you know you have a valid reference, so after you check the error.